### PR TITLE
Improve Activity Log page request lifecycle

### PR DESCRIPTION
**Pull request by balanza**
# Description

Refine how the Activity log page handles the API request
* define the HTTP request lifecycle in `lib/api/request`
* show an empty view initially
* show an error view in case an error occurred
* add a spinner when loading data

### CODE REVIEW

File: assets/js/pages/ActivityLogPage/ActivityLogPage.jsx [hunk 1]
```diff
@@ -103,19 +156,21 @@ function ActivityLogPage() {
...
-          data: data?.data || [],
-          pagination: data?.pagination,
-        });
-      })
-      .catch(() => setActivityLogResponse(emptyResponse))
-      .finally(() => {
-        setLoading(false);
-      });
+    // defer the loading state to avoid flickering
+    const tid = setTimeout(() => setActivityLogRequest(request.loading()), 500);
```
Comment (nelsonkopliku) [inline assets/js/pages/ActivityLogPage/ActivityLogPage.jsx:160]: **question:** how can the flickering be reproduced?

**Review by nelsonkopliku — COMMENTED** [review]
Interesting the introduction of the `MainView` and the modeling of the lifecycle of an API request. Curious about the possibility to be reused elsewhere.

However tests for `ActivityLogOverview` and `ActivityLogPage` are failing and need to be adjusted.

File: assets/js/pages/ActivityLogPage/ActivityLogPage.jsx [hunk 2]
```diff
@@ -52,19 +52,72 @@ const applyDefaultItemsPerPage = (params) =>
...
+  }
+  if (status === 'loading') {
+    return (
+      <div className="flex flex-row min-h-screen justify-center items-center">
+        <Spinner size="xl" />
+      </div>
+    );
+  }
+  if (status === 'failure') {
+    return <div>An error occurred.</div>;
```
Comment (nelsonkopliku) [inline assets/js/pages/ActivityLogPage/ActivityLogPage.jsx:78]: **suggestion:** we might want to style a tiny bit this case.

<details>
  <summary>See a screenshot here</summary>
  
image
</details>

File: assets/js/pages/ActivityLogPage/ActivityLogPage.jsx (hunk 1 above)
Comment (balanza) [reply assets/js/pages/ActivityLogPage/ActivityLogPage.jsx:160]: Good question, it is something visually annoying that has no functional effect. Maybe it only applies to local executions, when the latency is near zero.

File: assets/js/pages/ActivityLogPage/ActivityLogPage.jsx (hunk 2 above)
Comment (balanza) [reply assets/js/pages/ActivityLogPage/ActivityLogPage.jsx]: I reused the component from the other pages

image

Comment (nelsonkopliku) [reply assets/js/pages/ActivityLogPage/ActivityLogPage.jsx]: Awesome, thanks!

File: assets/js/pages/ActivityLogPage/ActivityLogPage.jsx (hunk 1 above)
Comment (nelsonkopliku) [reply assets/js/pages/ActivityLogPage/ActivityLogPage.jsx:179]: Alright thanks for clarifying.

**Review by nelsonkopliku — APPROVED** [review]
Besides the conflict to be resolved (my bad :smile: ) LGTM! Thanks!

**Pull request merged**